File: intent/src/main/java/com/xuebinduan/intent/MainActivity.java
package com.xuebinduan.intent;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.xuebinduan.intent.pictures.PictureChooseActivity;


/**
 * todo registerForActivityResult 使用这个需要引入最新的appcompat包，同时它需要在Started生命周期之前如onCreate中被注册。
 */
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        startActivity(new Intent(this, PictureChooseActivity.class));
    }


}

File: viewpager2/src/main/java/com/xuebinduan/viewpager2/MainActivity.kt
package com.xuebinduan.viewpager2

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /**
         * adapter:
         * FragmentStateAdapter
         * RecyclerView.Adapter
         */

        startActivity(Intent(this,ViewPager2BottomNavigationActivity::class.java))
//        startActivity(Intent(this,ViewPager2TabLayoutActivity::class.java))
//        startActivity(Intent(this,ViewPager2CustomTabActivity::class.java))

    }

}